src/common/Thread.cc

```diff
@@ -64,18 +64,17 @@ void *Thread::_entry_func(void *arg) {
 void *Thread::entry_wrapper()
 {
   int p = ceph_gettid(); // may return -ENOSYS on other platforms
-  if (p > 0)
+  if (p > 0) {
     pid = p;
-  if (pid &&
-      ioprio_class >= 0 &&
-      ioprio_priority >= 0) {
-    ceph_ioprio_set(IOPRIO_WHO_PROCESS,
-		    pid,
-		    IOPRIO_PRIO_VALUE(ioprio_class, ioprio_priority));
+    if (ioprio_class >= 0 &&
+	ioprio_priority >= 0) {
+      ceph_ioprio_set(IOPRIO_WHO_PROCESS,
+	  pid,
+	  IOPRIO_PRIO_VALUE(ioprio_class, ioprio_priority));
+    }
+    if (cpuid >= 0)
```

Comment:
i think we can just remove this check, as `_set_affinity()` is a no-op if `cpuid < 0`, and there is no need to put this call in the `if (p > 0)`, as `sched_setaffinity()` should be able to take care of itself as long as `HAVE_SCHED`.

also, no need to pass `pid` to `ceph_ioprio_set()`,

>               who is a process ID or thread ID identifying a single process or thread.  If who is 0, then operate on the calling thread.

per IOPRIO_SET(2).

no need to keep `pid` as a member variable anymore. all of its consumer are removed in this commit.